src/connectors/voice-search/connectVoiceSearch.ts

```diff
@@ -77,6 +90,20 @@ const connectVoiceSearch: VoiceSearchConnector = (
           const setQueryAndSearch = (query: string) => {
             if (query !== helper.state.query) {
               previousQuery = helper.state.query;
+              if (typeof additionalQueryParameters === 'function') {
+                const queryLanguages = language
+                  ? [language.split('-')[0]]
+                  : undefined;
+                helper.setState(
+                  helper.state.setQueryParameters({
+                    queryLanguages,
+                    ignorePlurals: true,
```

Comment:
@Haroenv 
When these three properties need to be set? When there's queryLanguages? When there's additionalQueryParams? or `queryLanguages || additionalQueryParams`?

```
                    ignorePlurals: true,
                    removeStopWords: true,
                    optionalWords: query,
```

Reply:
when additionalParameters is defined

Reply:
Can we replace 

```js
if (typeof additionalQueryParameters === 'function') {
  const queryLanguages = language
    ? [language.split('-')[0]]
    : undefined;
  helper.setState(
    helper.state.setQueryParameters({
      queryLanguages,
      ignorePlurals: true,
      removeStopWords: true,
      optionalWords: query,
      ...additionalQueryParameters({ query }),
    })
  );
}
```

with

```js
const queryLanguages = language
  ? [language.split('-')[0]]
  : undefined;

if (typeof queryLanguages !== 'undefined' || typeof additionalQueryParameters === 'function') {
  helper.setState(
    helper.state.setQueryParameters({
      queryLanguages,
      ...(typeof additionalQueryParameters === 'function' && {
        ignorePlurals: true,
        removeStopWords: true,
        optionalWords: query,
        ...additionalQueryParameters({ query }),
      })
    })
  )
}
```

Since `language` is a widget param, user should be able to use it without having to use `additionalQueryParameters`. I don't know how to make it prettier, though.